### DISCUSSION TITLE
(Lib/Go) Should I update to `v1.37.0`?

### DISCUSSION THREAD

**Discussion by psmarcin** [svix/svix-webhooks#1468] [Q&A] [answered]
Hey, 
thanks so much for all your hard work to create Svix SDK for Go. I've been using it for some time and it's super stable! 

Recently I had an issue while upgrading to version `v1.36.0`. It looks like there was breaking change introduced in that version, so I had to use pointer for `FilterTypes`. 

Today I noticed that you've released `v1.37.0` and there is another breaking change. 

It would be great to follow convention and bump major version when there is braking change included in release. 

The question is should I update to `v1.37.0` or wait some time to stabilize releases?

**Comment by tasn** [comment, marked as answer]
Hi @psmarcin, thanks for reaching out, and sorry for the trouble this has caused.

We follow a slightly modified semver for the SDKs. We don't break API as a rule (and if we ever will, we'll bump a major version) except for when breaking the API fixes a bug (as in: the API should never have been what it was, it was a mistake) and the fixes are extremely simple (you can add a * or a & or something like that to fix it). We won't break things in ways that change behavior or force you to make more difficult decisions than just fixing compilation errors.
This only applies to the SDKs, the HTTP API is not changing even if fixing bugs as that will actually break integrations (rather than being annoying when updating a version).

I know this is suboptimal, I really wish we didn't have to do it, but it's much better than the alternative. Consider this specific case, `FilterTypes` is nullable, that's actually a very legal (and often used) value for it. The Go SDK was deficient that it didn't allow that (this was a regression btw, it used to allow that), and doing this small break was the easiest way to fix it for everyone.

As for your question regarding updating: we found this whole class of bugs (it's in the OpenAPI generator that we utilize) which made all of these non-nullable. We fixed it in two spots for 1.36 and 1.37, but we'll also have a more sweeping fix in 1.38. After that, there are no immediate plans for anything else.

Please let me know what you think!

**Reply by psmarcin** [reply]
Thanks so much for extremely detailed answer. Now it seams clear for me what to do. Thanks again. 

Love your work, keep it up!

**Reply by tasn** [reply]
Thanks! 😊
